eth/api.go

```diff
@@ -478,7 +478,9 @@ func (api *PrivateDebugAPI) TraceTransaction(ctx context.Context, txHash common.
 		deadlineCtx, cancel := context.WithTimeout(ctx, timeout)
 		go func() {
 			<-deadlineCtx.Done()
-			tracer.(*ethapi.JavascriptTracer).Stop(&timeoutError{})
+			if deadlineCtx.Err() == context.DeadlineExceeded {
+				tracer.(*ethapi.JavascriptTracer).Stop(&timeoutError{})
+			}
```

Comment:
It's still not right. Err returns one of two values, context.Canceled or context.DeadlineExceeded. The change you made flips the check around but the semantics are the same.

The VM needs to be interrupted in both cases. I don't see a problem with ignoring the Err return value and just calling Stop. The interrupt channel is buffered and will just swallow the send if the VM is not running.